### PR TITLE
perf!: make the plugin compatible with Gradle's configuration cache

### PR DESCRIPTION
**Pull request by maksym-moroz**
TL;DR 
This PR addresses #676 and makes [git-sensitive-semantic-versioning-gradle-plugin](https://github.com/DanySK/git-sensitive-semantic-versioning-gradle-plugin) compatible with configuration cache.

Gradle [configuration cache](https://docs.gradle.org/current/userguide/configuration_cache.html) allows to significantly reduce configuration time and improve build performance by caching results of configuration phase.
There is a [list of requirements](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements) to adhere for configuration cache compatibility.
This PR adresses two separate issues with previous implementation:
- Usage of `Runtime.exec()` for handling external processes 
- Isolation of `Project` instance during configuration

>I had an idea of enabling context receivers to pass down implicit dependencies introduced by lifting Project but it looks like context receivers are not compatible with configuration cache due to a [bug](https://github.com/gradle/gradle/issues/27725) so instead I opted for moving runCommand suite functions into extension for easier access to dependencies needed.

Changes were extensively tested locally.

Additional thing to consider: PR changes the contract of `assignGitSemanticVersion()` function from internal mutation of the project.version to returning the computed value without mutation. 
Since `assignGitSemanticVersion()` is a part of the public API this introduces a **BREAKING CHANGE** for consumers who rely on the side effect nature of original `assignGitSemanticVersion()` to update project.version.

### CODE REVIEW

**Comment by maksym-moroz** [comment]
Special thanks to @AirCatche for getting to bottom of this together.

**Review by DanySK — APPROVED** [review]

**Comment by DanySK** [comment]
This is a great contribution and I am approving it.
